File: perlfuck.pl
#!/usr/bin/perl
use strict;
use warnings;

my $one = "(()=({}.{})=~/()/)";
my $zero = "(()=({})=~/^\$/)";

sub number {
	my ($n) = @_;

	if ($n == 0) {
		return $zero;
	}

	my @expression = ($one) x $n;
	return join("+", @expression);
}

my %map;

sub from_string {
	my ($s) = @_;
	my @chars = split("", $s);
	my @encoded_chars;

	my $last_index = $#chars;
	my $loop_counter = eval number(0);

	foreach my $char (@chars) {
		my $char_code = ord($char);
		push @encoded_chars, "chr(" . number($char_code) . ")";
		if ($loop_counter < $last_index) {
			push @encoded_chars, ".";
		}
		$loop_counter += eval number(1);
	}
	return join("", @encoded_chars);
}

print "eval(" . from_string('print "test";') . ");";

File: perlfuck.js
const one = "(()=({}.{})=~/()/)";
const zero = "(()=({})=~/^$/)";

function number(n) {
    if (n === 0) {
        return 0;
    }

    const expression = new Array(n).fill(one);
    return expression.join("+");
}

function fromString(s) {
    const chars = s.split("");
    const encodedChars = [];
    let loopCounter = 0;
    chars.forEach((char, index) => {
        const charCode = char.charCodeAt(0);
        encodedChars.push(`chr(${number(charCode)})`);
        if (loopCounter < chars.length - 1) {
            encodedChars.push(".");
        }
        loopCounter += 1;
    });

        return "eval(" + encodedChars.join("") + ");";
}

function convertCode() {
        const perlCode = document.getElementById("perlCode").value;
        const convertedCode = fromString(perlCode);
        document.getElementById("perlfuckCode").value = convertedCode;
}
